src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -258,6 +258,17 @@ protected override IEnumerable<TypeInferenceInfo> InferTypesWorker_DoNotCallDire
                     YieldStatementSyntax yieldStatement => InferTypeInYieldStatement(yieldStatement, token),
                     _ => SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>(),
                 };
+
+                var rightToken = syntaxTree.FindTokenOnRightOfPosition(position, CancellationToken);
+
+                // If we are before the `..` token of a range expression, add `int` to infered types if it is not already there
+                if (rightToken is { RawKind: (int)SyntaxKind.DotDotToken, Parent: RangeExpressionSyntax } &&
+                    !inferedInfos.Contains(t => t.InferredType.SpecialType == SpecialType.System_Int32))
+                {
+                    inferedInfos = inferedInfos.Concat(new TypeInferenceInfo(Compilation.GetSpecialType(SpecialType.System_Int32)));
+                }
```

Comment:
do we need this.  almost *always* we don't consider waht follows the location as it often may be in some indeterminate state.   the test case for this isn't very compelling either.  i don't think it's necessary to show things like `int.maxvalue` there.  `int` itself should show up in the list and be good enough.

Reply:
I did it mostly for consistency. If we show these items in `..$$` scenario, I think we should show them in `$$..` as well. If you don't accept prefix scenario, I will just revert the last commit at all

Reply:
yes, please revert.  thansk :)